ddtrace/bootstrap/sitecustomize.py

```diff
@@ -74,6 +75,75 @@ def update_patched_modules():
         EXTRA_PATCHED_MODULES[module] = asbool(should_patch)
 
 
+if PY2:
+    _unloaded_modules = []
+
+
+def is_installed(module_name):
+    # https://stackoverflow.com/a/51491863/735204
+    if sys.version_info >= (3, 4):
+        return importlib.util.find_spec(module_name)
+    elif sys.version_info >= (3, 3):
+        return importlib.find_loader(module_name)
+    elif sys.version_info >= (3, 1):
+        return importlib.find_module(module_name)
+    elif sys.version_info >= (2, 7):
+        try:
+            imp.find_module(module_name)
+        except ImportError:
+            return False
+        else:
+            return True
+    return False
+
+
+def should_cleanup_loaded_modules():
+    dd_unload_sitecustomize_modules = os.getenv("DD_UNLOAD_MODULES_FROM_SITECUSTOMIZE", default="auto").lower()
+    if dd_unload_sitecustomize_modules == "0":
+        log.debug("DD_UNLOAD_MODULES_FROM_SITECUSTOMIZE==0: skipping sitecustomize module unload")
+        return False
+    elif dd_unload_sitecustomize_modules not in ("1", "auto"):
+        log.debug(
+            "DD_UNLOAD_MODULES_FROM_SITECUSTOMIZE=={}: skipping sitecustomize module unload because of invalid value",
+            dd_unload_sitecustomize_modules,
+        )
+        return False
+    elif dd_unload_sitecustomize_modules == "auto" and not any(
+        is_installed(module_name) for module_name in MODULES_THAT_TRIGGER_CLEANUP_WHEN_INSTALLED
+    ):
+        log.debug(
+            "DD_UNLOAD_MODULES_FROM_SITECUSTOMIZE==auto: skipping sitecustomize module unload because "
+            "no module requiring unloading is installed"
+        )
+        return False
+    return True
+
+
+def cleanup_loaded_modules_if_necessary():
+    if not should_cleanup_loaded_modules():
+        return
+    modules_loaded_since_startup = set(_ for _ in sys.modules if _ not in MODULES_LOADED_AT_STARTUP)
+    modules_to_cleanup = modules_loaded_since_startup - MODULES_TO_NOT_CLEANUP
+    # Unload all the modules that we have imported, except for ddtrace and a few
+    # others that don't like being cloned.
+    # Doing so will allow ddtrace to continue using its local references to modules unpatched by
+    # gevent, while avoiding conflicts with user-application code potentially running
+    # `gevent.monkey.patch_all()` and thus gevent-patched versions of the same modules.
+    for m in modules_to_cleanup:
+        if any(m.startswith("%s." % module_to_not_cleanup) for module_to_not_cleanup in MODULES_TO_NOT_CLEANUP):
```

Comment:
🤔 don't we need to also `m == module_to_not_cleanup` to check for the root module?